### ISSUE TITLE
Add warning if Custom Location checkbox is selected

### ISSUE THREAD

**Issue by sbabur** [disarm-platform/user-requests-and-feedback#59]
We ran into issues last year where users were needlessly using the Location checkbox and typing the village instead of selecting it from the dropdown, which caused a lot of confusion on the dashboard.

We are going to hit this point hard in the trainings, but we were also thinking it would be good to add a single confirmation mechanism where the user has to click ok in order to proceed with text:

"Are you sure this place name does not fall within the sub-areas provided? If you proceed with this other name, your data will be saved, but not appear in the dashboard"

**Comment by onlyjsmith** [comment]
Ok - we’ll have a think about the options as well.

**Comment by onlyjsmith** [comment]
@sbabur We've got two options, neither included in the 1.2.9 version - but either could be included tomorrow (Weds June 20) morning, to still give us time to test. I prefer the suggestion one, but the other is definitely simpler (and more annoying and less useful ;-) )

pop-up message when checkbox is clicked: https://deploy-preview-374--douma-app.netlify.com/#instance=nam

shows suggestions when custom location name is typed: https://deploy-preview-373--douma-app.netlify.com/#instance=nam

**Comment by sbabur** [comment]
@onlyjsmith Thanks for letting me test out both options. Option 2 is really cool and I think immensely helpful! Could we go with that one and deploy it tomorrow? 

Would it be possible to still add a text note below **IF** a custom selection is ultimately chosen though? Let me know if this is something that can be done quickly enough in time to release by tomorrow or not. 

"Note: data with this custom location will be recorded, but will not appear on the dashboard"

![image](https://user-images.githubusercontent.com/37877007/41612898-7345223c-73f4-11e8-929e-131774e6b6e4.png)

![image](https://user-images.githubusercontent.com/37877007/41612938-83727196-73f4-11e8-9db8-522203a867e4.png)

**Comment by onlyjsmith** [comment]
@sbabur very glad you like 2 - we do too! Unfortunately we uncovered a couple UI issues in our testing, that we think mean it would be more confusing to include it at this point. 

So, suggest we stick with the simpler pop-up version for now, and see if we can find time to work on the little UI issues.

**Comment by sbabur** [comment]
@onlyjsmith Can you describe what the UI issues were? I failed to find any, I thought it was super smooth on both desktop and mobile but good to know.

Fine with using the pop-up. Would be a huuuge nice to have if you manage to address the UI issues though.

**Comment by onlyjsmith** [comment]
There’s a gap in the user flow, where it’s not necessarily obvious you need to click on the orange box to select it when it’s a real custom entry. It means we might actually make it harder to create a valid form when adding a custom text entry. Since the custom text entry is already off the main flow we want a user to follow (i.e. for some reason they can’t find what they think they should find in the dropdown), we need to make sure it’s simple and reliable (even with the pop-up to confirm).

It’s definitely possible to work around, but it just requires more thinking, and might require some changes further up. Then pretty quickly we get to this ‘custom entry' being a better experience than the existing main double-dropdown - at that points we’d really like to replace the existing solution with this new single text box. We’re unlikely to get it done in time for next week. We’ll take another look tomorrow though.

**Comment by sbabur** [comment]
Yeah, the point about this being a better location selection overall occurred to me as well. 

To your first point about valid form, I think this relates to a separate issue we've worked around which is that the error validations only prevent you from proceeding AFTER page 1. Ideally, a blank GPS coordinate or Location selection would grey out the Next button like the other pages. 

Fine to table. I will pass the testing link to Ayo/Katelyn/Lakshmi/Dennis - curious as to their reception. I think locations are pretty straightforward in Botswana, but something like this could be immensely helpful in an instance like Namibia. Then again, they may get the spelling wrong, or get confused why so many places that are way way way outside their spray area are appearing, since planning and implementation are so decentralized.

**Comment by sbabur** [comment]
@onlyjsmith I talked to Katelyn about this because she has a lot more experience in Namibia on admin units. My sense was that users in Namibia sometimes ONLY know the village so the cascading dropdown was difficult for them, whereas in Botswana it makes a lot more sense because they know their districts really well and don't want to bother looking at villages from elsewhere. 

Katelyn - "I really love this. It is absolutely the case that data collectors often do not know the constituency of a village, or that the village places in the wrong constituency (because we have it in the wrong place in our shapefile).  For the ‘custom location’ we should certainly go with this design if we are not already. 
Then regarding preference to always type the name, I would think that – yes – this is the easier option, but I wonder if we could poll this in Botswana next week?"

We could do some user testing on this in Botswana? I was also thinking that we have to choose between training issues for either approach, and for the sake of data quality, it is easier and better to train them on that orange box. While it is definitely a better experience overall for Namibia, that might not necessarily be the case in Botswana, which is why we are happy to keep it only as part of the custom checkbox workflow and not the dropdown one. 

In summary: can we do user testing to confirm, and then deploy Option 2 as is next week, and just train around the UI kinks? We think it will make a huge difference on the Namibia dashboard.
